gcore/src/msg.rs

```diff
@@ -190,10 +190,19 @@ pub fn with_read_on_stack<T>(f: impl FnOnce(Result<&mut [u8]>) -> T) -> T {
         let mut len = 0u32;
 
         if size > 0 {
-            unsafe { gsys::gr_read(0, size as u32, buffer.as_mut_ptr(), &mut len as *mut u32) }
+            unsafe {
+                gsys::gr_read(
+                    0,
+                    size as u32,
+                    buffer.as_mut_ptr() as *mut u8,
+                    &mut len as *mut u32,
+                )
+            }
         }
 
-        f(SyscallError(len).into_result().map(|_| buffer))
+        f(SyscallError(len)
+            .into_result()
+            .map(|_| unsafe { &mut *(&mut buffer[..size] as *mut _ as *mut [u8]) }))
```

Comment:
what happened here? xd

Reply:
this is https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#method.slice_assume_init_ref

Reply:
I also checked it with https://github.com/rust-lang/rust-bindgen
```
$ bindgen alloca_.h

...

extern "C" {
    pub fn c_with_alloca(
        size: usize,
        callback: ::std::option::Option<
            unsafe extern "C" fn(arg1: *mut u8, arg2: *mut ::std::os::raw::c_void),
        >,
        data: *mut ::std::os::raw::c_void,
    );
}
```